torch/testing/_internal/common_utils.py

```diff
@@ -2100,6 +2101,13 @@ def assertRaises(self, expected_exception, *args, **kwargs):
     # Reimplemented to provide special behavior when
     # _ignore_not_implemented_error is True
     def assertRaisesRegex(self, expected_exception, expected_regex, *args, **kwargs):
+        # Check with hasattr as not all Test classes have attribute `device_type`.
```

Comment:
I think this comment could be tweaked for readability.

First, let's add a comment to `assertRaisesRegex` describing what it does:

> Verifies that an exception with the type `expected_exception` and message matching the regular expression defined by `expected_regex` is thrown. If the test is instantiated for a non-native device type (like XLA) then the message is not validated.

Then before the `hasattr` let's clarify what's up:

> Checks whether the test is instantiated for a device type by testing if the test class has defined the `device_type` attribute and, if so, tests whether the instantiated device type is native or not

Reply:
Done :). Thanks!